### DISCUSSION TITLE
Use GatsbyImage component from gatsby-plugin-image in zoomImg props

### DISCUSSION THREAD

**Discussion by lezan** [rpearce/react-medium-image-zoom#349] [Ideas]
Hello @rpearce, just an idea about a new feature to extend the compatibility with the `gatsby-plugin-image`. It would be cool if it were possibility to use a `GatbsyImage` component as `zoomImg` props.

Something like:

```
<Zoom
  zoomImg={
    <GatsbyImage
      className="image-zoom"
      alt={alt}
      image={getImage(large)}
    />
  }
>
  <GatsbyImage
    className="image-zoom"
    alt={alt}
    image={getImage(small)}
  />
</Zoom>
```

**Comment by rpearce** [comment]
@lezan Can you help me understand what this buys you over passing the `src`, `sizes`, `srcset`, and other image props, oneself? At present this should work (I'm assuming `getImage(large)` returns a `src`):

```js
<Zoom zoomImg={{ alt, src: getImage(large) }}>
  <GatsbyImage
    className="image-zoom"
    alt={alt}
    image={getImage(small)}
  />
</Zoom>
```

In case you didn't know: I recommend styling things by either overriding the selectors in https://github.com/rpearce/react-medium-image-zoom/blob/main/source/styles.css or by copying the CSS file and adding your own values.

**Reply by lezan** [reply]
Hello @rpearce,
do you mean pass `GatsbyImage` component instead of `src` and so on? I am not sure, but intuitively nothing if you rebuild the same object (placeholder, lazy loading, fallback, srcset and sizes).

I tried your example, but it seems the image pass in `zoomImg` is not used, but instead the same of low quality declare inside `Zoom` component (small instead of large).

**Reply by rpearce** [reply]
@lezan does your test example have your zoomIng attempt? I can look at it and see what's going on. That feature works (at least with my examples), but there might be something to adjust on your side.

**Reply by lezan** [reply]
@rpearce let me update it for you.

Is it correct? https://github.com/rpearce/react-medium-image-zoom/blob/main/source/Controlled.tsx#L207. I can be wrong, but are you using the right `src` or the one inside the `Zoom` component? Because here https://github.com/rpearce/react-medium-image-zoom/blob/main/source/Controlled.tsx#L138 you are checking the `zoomImg` `src`. If you get what I mean.

EDIT: updated https://github.com/lezan/zoom-test-gatsby

**Reply by rpearce** [reply]
@lezan If you look a couple of lines down, https://github.com/rpearce/react-medium-image-zoom/blob/3ee3e98d6651b1975b49292f436f7915829cf5b1/source/Controlled.tsx#L212, the `zoomImg` properties override the ones that come before it. The `zoomImg`' `src` is loaded in the background: https://github.com/rpearce/react-medium-image-zoom/blob/3ee3e98d6651b1975b49292f436f7915829cf5b1/source/Controlled.tsx#L448-L469

**Reply by lezan** [reply]
@rpearce I was wrong!

If you wanna give a try:
- https://codesandbox.io/s/silent-platform-o3ci0s?file=/src/pages/index.js
- https://github.com/lezan/zoom-test-gatsby

From my side, there is always the same (smaller) image.

EDIT: could be related to the override at https://github.com/rpearce/react-medium-image-zoom/blob/3ee3e98d6651b1975b49292f436f7915829cf5b1/source/Controlled.tsx#L212? Because `sizes`, `src` and `srcset` are the same.

**Reply by lezan** [reply]
@rpearce removing into `Zoom` component the `gatsby-plugin-image` and adding `<img />` work as expected. The issue seems to be related to how to make working the override with the `StaticImage` and `GatsbyImage`, I think.

**Reply by rpearce** [reply]
@lezan ok, I will check out your repo tonight and see what I can do.

**Reply by rpearce** [reply]
@lezan Here's a pull request for you: https://github.com/lezan/zoom-test-gatsby/pull/2

**Reply by lezan** [reply]
@rpearce work like a charm!
